### ISSUE TITLE
BUG on pypi deploy

### ISSUE THREAD

**Issue by mdantonio** [rapydo/issues#100]
All tools (except for http-api) are tested on several python version and each job then deploys on pypi => 2 jobs out of 3 fails with the error:

```
PyPI upload failed.
failed to deploy
```

I guess that the new pypi makes now this verification

Examples here:

https://travis-ci.org/rapydo/utils

https://travis-ci.org/rapydo/do

https://travis-ci.org/rapydo/develop

To fix:

remove deploy from the main jobs, add a further step with deploy only (as for [http-api](https://github.com/rapydo/http-api/blob/master/.travis.yml))

**Comment by pdonorio** [comment]
I currently see `Skipping a deployment with the pypi provider because this is not a tagged commit` 
https://travis-ci.org/rapydo/utils/jobs/400779872#L851

It this problem still existing?

**Comment by mdantonio** [comment]
Yes, but only occurs on deployment (i.e. when the commit is tagged) like here:

https://travis-ci.org/rapydo/utils/builds/371532623

You linked a normal commit

The problem affects utils do end develop because all test against 3 different python version and the deploy step is at the same level of these => is repeated three time (the first success successes, the secondo and the third fails)

The problem do not affect http api because it defines different stages and deploy is called once

**Comment by pdonorio** [comment]
Why aren't the different python versions executed in parallel as different stages?
I can't really remember how this work on travis,
but if I understand correctly we need to launch the pypi submission when the three versions checks are done.

BTW: we need to add `python 3.7.0` and make versions checks 4 :)

**Comment by mdantonio** [comment]
> Why aren't the different python versions executed in parallel as different stages?

They are, the problem is deploy that should be executed only once

The solution should be introduce stages as in http api.

The difference is: http api do stages against different ENV variables. Here we have stages against different Python versions

**Comment by mdantonio** [comment]
Travis does not support python 3.7

We can add 3.7-dev or wait for the release

**Comment by mdantonio** [comment]
Applied the same change to do and develop

Added 3..7-dev but it fails with a gcc error:

https://travis-ci.org/rapydo/do/builds/403091892

Do you like to debug 3.7-dev? Otherwise i will remove it and back the our three version

**Comment by pdonorio** [comment]
useless to debug `3.7-dev` when the stable version is out 😉

**Comment by mdantonio** [comment]
Removed, just waiting for travis to add 3.7 (hope in a near future)

Here the list of supported versions:

https://docs.travis-ci.com/user/languages/python/

Waiting for the next release to verify the deploy step

**Comment by mdantonio** [comment]
Deploy fails for all tools (utils, http-api, do) but with different messages.

Starting with do, since here the problem can be easly fixed:

```
fatal: Not a git repository (or any of the parent directories): .git
PyPI upload failed.
failed to deploy
```

Removed from .travis:

```
before_deploy:
# /home/travis/build/rapydo/do
- cd .. && pwd
```

(other tools have not this piece of code)

**Comment by mdantonio** [comment]
Authenticated as pdonorio
No stash entries found.
PyPI upload failed.
failed to deploy

@pdonorio any idea on how to debug?

**Comment by mdantonio** [comment]
I found the problem, it was hidden inside the `Deploying application` section (closed by default)

Here the error:

`HTTPError: 400 Client Error: User 'pdonorio' does not have a verified primary email address. Please add a verified primary email before attempting to upload to PyPI. See https://pypi.org/help/#verified-email for more information.for more information. for url: https://upload.pypi.org/legacy/`

**Comment by pdonorio** [comment]
Verified my email

**Comment by mdantonio** [comment]
It worked with develop

Waiting for utils, controller and http-api
